contracts/supplementary/rebalancing-manager/ETHDaiRebalancingManager.sol

```diff
@@ -0,0 +1,478 @@
+import { ICore } from "../../core/interfaces/ICore.sol";
+import { IMedian } from "../../external/DappHub/interfaces/IMedian.sol";
+import { IRebalancingSetToken } from "../../core/interfaces/IRebalancingSetToken.sol";
+import { ISetToken } from "../../core/interfaces/ISetToken.sol";
+import { RebalancingHelperLibrary } from "../../core/lib/RebalancingHelperLibrary.sol";
+
+
+/**
+ * @title ETHDaiRebalancingManager
+ * @author Set Protocol
+ *
+ * Contract used to manage a ETHDai Rebalancing Set Token
+ */
+contract ETHDaiRebalancingManager {
+
+    using SafeMath for uint256;
+    using AddressArrayUtils for address[];
+
+    /* ============ Constants ============ */
+
+    uint256 constant PRICE_PRECISION = 100;
+    uint256 constant AUCTION_LIB_PRICE_DIVISOR = 1000;
+    uint256 constant DAI_PRICE = 10**18;
+    uint256 constant DAI_DECIMALS = 18;
+    uint256 constant ETH_DECIMALS = 18;
+    uint256 constant SET_TOKEN_DECIMALS = 18;
+    uint256 constant THIRTY_MINUTES_IN_SECONDS = 1800;
+    uint256 constant VALUE_TO_CENTS_CONVERSION = 10**16;
+
+
+    /* ============ State Variabales ============ */
+
+    address public daiAddress;
+    address public ethAddress;
+    address public setTokenFactory;
+
+    address public ethPriceFeed;
+
+    address public coreAddress;
+
+    address public auctionLibrary;
+    uint256 public auctionTimeToPivot;
+    uint256 public daiMultiplier;
+    uint256 public ethMultiplier;
+
+    uint256 public maximumLowerThreshold;
+    uint256 public minimumUpperThreshold;
+
+    /* ============ Events ============ */
+
+    event LogManagerProposal(
+        uint256 ethPrice
+    );
+
+    /* ============ Constructor ============ */
+
+    /*
+     * Rebalancing Token Manager constructor.
+     * The multipliers are used to calculate the allocation of the set token. Allocation
+     * is determined by a simple equation:
+     *      daiAllocation = daiMultiplier/(daiMultiplier + ethMultiplier)
+     * Furthermore the total USD cost of any new Set Token allocation can be found from the
+     * following equation:
+     *      SetTokenUSDPrice = (daiMultiplier + ethMultiplier)*max(ethPrice, daiPrice)
+     *
+     * @param  _coreAddress             The address of the Core contract
+     * @param  _ethPriceFeedAddress     The address of ETH medianize
+     * @param  _daiAddress              The address of the Dai contract
+     * @param  _ethAddress              The address of the wrapped ETH contract
+     * @param  _setTokenFactory         The address of the SetTokenFactory
+     * @param  _auctionLibrary          The address of auction price curve to use in rebalance
+     * @param  _auctionTimeToPivot      The amount of time until pivot reached in rebalance
+     * @param  _multipliers             Token multipliers used to determine allocation
+     * @param  _allocationBounds        Bounds to stop proposal if not enough deviation from expected allocation
+     */
+    constructor(
```

Comment:
Can we DRY this up similar to how its done with `ModuleCoreState` and the modules?

Reply:
Gonna do this when we generalize it